src/screens/grocerylist/GroceryList.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect } from 'react';
+import { StyleSheet, Dimensions, ScrollView } from 'react-native';
+import { Box, Text } from '../../components/index';
+import Svg, { G, Path, Rect, TSpan } from "react-native-svg";
+import IconButton from '../../components/IconButton';
+import GroceryItem from "./GroceryItem";
+import { withGroceryListService } from '../../services'
+import { FoodItem } from "../../models/FoodItem"
+import { logError } from '../../logger/Logger';
+
+const windowWidth = Dimensions.get('window').width;
+const windowHeight = Dimensions.get('window').height;
+
+const styles = StyleSheet.create({
+    groceryListName: {
+        width: windowWidth - 64,
+        height: 35,
+        marginTop: 18,
+        marginLeft: 32
+    },
+    groceryList: {
+        height: windowHeight - 315,
+        marginTop: 40,
+        marginLeft: 32,
+        flexDirection: "row"
+    },
+    item: {
+        marginLeft: 16,
+        marginBottom: 20
+    },
+    addItem: {
+        marginLeft: 45,
+        marginTop: 10
+    },
+    container: {
+        height: windowHeight
+    }
+});
+
+const GroceryList = () => {
+
+    const { groceryListService } = withGroceryListService()
+
+    const [todos, setTodos] = useState([]);
+    const [grocList, setGrocList] = useState([]);
+
+    async function fetchGroceryList() {
+        let gList = await groceryListService.getGroceryListById("ReyesGroceryList");
+        setGrocList(gList);
+        const foodItemArr = await groceryListService.getFoodItemArrayFromGroceryList("ReyesGroceryList");
+        setTodos(foodItemArr);
+    }
+
+    async function addTodo(text) {
+        try {
+            const itemToAdd = new FoodItem(text, "", "", "", new Date(), {});
+            let newList = await groceryListService.addFoodItemToGroceryList(grocList, itemToAdd);
+            setGrocList(newList);
+            const newTodos = [...todos, {text}];
+            setTodos(newTodos);
+        } catch (err) {
+            console.error(err);
+            logError(err);
+        }
+    };
+
+    async function updateTodo(text, index) {
+        try {
+            const itemToUpdate = new FoodItem(text, "", "", "", new Date(), {});
+            let newList = await groceryListService.updateFoodItemInGroceryList(grocList, index, itemToUpdate);
+            setGrocList(newList);
+        } catch (err) {
+            console.error(err);
+            logError(err);
+        }
+    }
+
+    async function removeTodo(index) {
+        try {
+            let newList = await groceryListService.removeFoodItemFromContainer(grocList, index);
+            setGrocList(newList);
+            const newTodos = [...todos];
```

Comment:
@ynigoreyes 

I'm fairly sure the problem is here. For now, I'm using the hamburger icon as a temporary delete button. So when you press that button it deletes the item from the list by passing the index of the item in the todos array to the removeTodo() function. The correct item is always deleted from the backend, but it only appears to work correctly on the front end when you delete the item at the bottom of the list. Basically, on the front end no matter what item you try to delete, it removes the item from the end of the list. I think the setTodos() call may not be working?

I've confirmed that the correct index is being passed so I'm probably just missing something!

Reply:
Why do we have a todo list and a groc list? Are you trying to keep two separate lists?

Reply:
groclist is a different object, the todo list is the array of foodItems within the grocery list